docs/prover/cli/options.md

```diff
@@ -369,17 +369,53 @@ certoraRun Bank.sol --verify Bank:Bank.spec --optimistic_loop
 ### `--loop_iter`
 
 **What does it do?**
-Sets the maximal number of loop iterations we verify for. The way the Certora Prover handles loops is by unrolling them - if the loop should be executed three times, it will copy the code inside the loop three times. This option sets the number of unrolls. Be aware that the run time grows exponentially by the number of loop iterations.  
+Sets the maximal number of loop iterations we verify for. The way the Certora Prover handles loops is by unrolling them - if the loop should be executed three times, it will copy the code inside the loop three times. This option sets the number of unrolls. Be aware that the run time grows exponentially by the number of loop iterations.
 
-**When to use it?**  
-The default number of loop iterations we unroll is one. However, in many cases, bugs only occur when there are several iterations. Common scenarios include iteration over list elements. Two, or in some cases three, is usually the most you will ever need to uncover bugs.  
+**When to use it?**
+The default number of loop iterations we unroll is one. However, in many cases, bugs only occur when there are several iterations. Common scenarios include iteration over list elements. Two, or in some cases three, is usually the most you will ever need to uncover bugs.
 
 **Example**
 
 ```
 certoraRun Bank.sol --verify Bank:Bank.spec --loop_iter 2
 ```
 
+Options regarding summarization
+-------------------------------
+
+(--optimistic_summary_recursion)=
+### `--optimistic_summary_recursion`
+
+**What does it do?**
+In case there's a call to some Solidity function within a summary, we may end up with recursive calls to this summary. For example, if in the summary of `foo` we call the Solidity function `bar`, and `bar`'s Solidity code contains a call to `foo`, we'll summarize `foo` again, which will lead to another call to `bar` etc.
+The default behavior in this case is to assert if the recursion limit is reached (the limit is controlled by the {ref}`--summary_recursion_limit` flag). Setting this flag to `true` will instead assume that the limit is never reached.
+
+**When to use it**
+There's recursion due to summaries calling Solidity functions, and this leads to an assertion failure due to this. In this case one can either make the limit larger (via {ref}`--summary_recursion_limit`) or set this flag to `true`.
```

Comment:
can we make this even easier to users - and say which assert message they will see? I believe it's this:
`Recursion limit ... for calls to ..., reached during compilation of summary ...`